### ISSUE TITLE
HA 2024.1.0 deprecations

### ISSUE THREAD

**Issue by tjorim** [sca075/valetudo_vacuum_camera#62]
### Checklist

- [X] I have updated the integration to the latest version available
- [X] I have checked if the problem is already reported

### The problem

Updated to HA version 2024.1.0b0 and saw the following warning:

```
Logger: homeassistant.components.camera
Source: helpers/deprecation.py:204
Integration: Camera (documentation, issues)
First occurred: 16:22:46 (6 occurrences)
Last logged: 16:22:48

SUPPORT_ON_OFF was used from valetudo_vacuum_camera, this is a deprecated constant which will be removed in HA Core 2025.1. Use CameraEntityFeature.ON_OFF instead, please create a bug report at https://github.com/sca075/valetudo_vacuum_camera/issues
```

Also, the following two errors:
```
Logger: homeassistant.components.camera
Source: helpers/entity_platform.py:507
Integration: Camera (documentation, issues)
First occurred: 16:22:47 (2 occurrences)
Last logged: 16:22:48

Error adding entities for domain camera with platform valetudo_vacuum_camera
Error adding entities for domain camera with platform tapo_control
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 507, in async_add_entities
    await asyncio.gather(*tasks)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 752, in _async_add_entity
    await entity.add_to_platform_finish()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1273, in add_to_platform_finish
    await self.async_internal_added_to_hass()
  File "/usr/src/homeassistant/homeassistant/components/camera/__init__.py", line 744, in async_internal_added_to_hass
    self._rtsp_to_webrtc = await self._async_use_rtsp_to_webrtc()
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/camera/__init__.py", line 761, in _async_use_rtsp_to_webrtc
    if CameraEntityFeature.STREAM not in self.supported_features:
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: argument of type 'int' is not iterable
```

and

```
Logger: homeassistant.components.camera
Source: helpers/entity_platform.py:368
Integration: Camera (documentation, issues)
First occurred: 16:22:47 (2 occurrences)
Last logged: 16:22:48

Error while setting up valetudo_vacuum_camera platform for camera
Error while setting up tapo_control platform for camera
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 368, in _async_setup_platform
    await asyncio.gather(*pending)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 507, in async_add_entities
    await asyncio.gather(*tasks)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 752, in _async_add_entity
    await entity.add_to_platform_finish()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1273, in add_to_platform_finish
    await self.async_internal_added_to_hass()
  File "/usr/src/homeassistant/homeassistant/components/camera/__init__.py", line 744, in async_internal_added_to_hass
    self._rtsp_to_webrtc = await self._async_use_rtsp_to_webrtc()
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/camera/__init__.py", line 761, in _async_use_rtsp_to_webrtc
    if CameraEntityFeature.STREAM not in self.supported_features:
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: argument of type 'int' is not iterable
```

### What version of an integration has described problem?

v1.5.2-alpha

### What vacuum model do you have problems with?

n/a

### Please firmware installed on your Vacuum.

n/a

### What version of Home Assistant do you use?

core-2024.1.0b0

### What type of installation are you running?

Home Assistant OS

**Comment by sca075** [comment]
Wow thanks @tjorim, I didn't got this error during the test as per I use the last released (not beta) version of HA. I will update the code according to this.. it will take few hours I think...

**Comment by tjorim** [comment]
No worries, thanks for your effort. Do you accept tips?

**Comment by sca075** [comment]
@tjorim of course, I don't to begin the new year with a bug that block everyone, and tips are alway welcome.. the last release of the camera, by the way is not the v1.5.2-alpha (there are great improvements on the current v1.5.3) please give it a try.. in few hours as I wrote I will publish the fix for this ;)

**Comment by tjorim** [comment]
Thanks, I just noticed and updated accordingly. Please let me know where I can tip.

**Comment by sca075** [comment]
Well man, for me the higher tip is knowing you use this and you like it ... :) 
 Would appreciate if you can tip Denis or Sören https://github.com/sponsors/Hypfer  
They are doing an amazing job with the Vacuums and need our support ;)

**Comment by sca075** [comment]
@tjorim  if now download the main (with HACS just re-download and then select the version "main" and wait for a second that it is selected), it should be fixed, thanks once again for the advice and head up for this :)

**Comment by tjorim** [comment]
All good and working, only noticed this warning in the logging now:

```
Logger: py.warnings
Source: /usr/local/lib/python3.11/warnings.py:109
First occurred: 18:44:54 (2 occurrences)
Last logged: 18:45:32

/usr/local/lib/python3.11/threading.py:982: RuntimeWarning: coroutine 'ValetudoCamera.async_update' was never awaited self._target(*self._args, **self._kwargs)
```

**Comment by sca075** [comment]
okay let me have a look at it... as soon the test instance will boot again :) I undated it to the same version you use.

**Comment by sca075** [comment]
@tjorim I'm sorry, I could not reproduce the issue with the async_update, if you enable the debug log of the valetudo vacuum camera and then please just reboot HA, the camera save the logs and then you can get them on the www folder from there you can download the logs via VS Code or File Editor or copy them with SAMBA. If you can do that.. then I will give you an email address were to send the logs.

**Comment by tjorim** [comment]
[Uploading home-assistant_2023-12-28T18-29-08.463Z.log…]()

**Comment by sca075** [comment]
@tjorim okay now should be fine, same procedure as before and the warning should be gone.

**Comment by tjorim** [comment]
Can confirm, all good!

**Comment by sca075** [comment]
Released just now v1.5.4 with this fix, once again thank a lot for the head up @tjorim
